pennylane_lightning/core/src/simulators/lightning_tensor/cutn/tests/Tests_CutnMPS.cpp

```diff
@@ -0,0 +1,158 @@
+    }
+}
+
+TEMPLATE_PRODUCT_TEST_CASE("MPSCutn::Constructibility",
+                           "[General Constructibility]", (MPSCutn),
+                           (float, double)) {
+    using MPST = TestType;
+
+    SECTION("MPST<TestType>") { REQUIRE(!std::is_constructible_v<MPST>); }
+    SECTION("MPST<TestType> {const size_t, const size_t, DevTag<int> &}") {
+        REQUIRE(std::is_constructible_v<MPST, const size_t, const size_t,
+                                        DevTag<int> &>);
+    }
+}
+
+TEMPLATE_TEST_CASE("MPSCutn::SetBasisStates() & reset()", "[MPSCutn]", float,
+                   double) {
+    std::vector<std::vector<size_t>> basisStates = {
+        {0, 0, 0}, {0, 0, 1}, {0, 1, 0}, {0, 1, 1},
+        {1, 0, 0}, {1, 0, 1}, {1, 1, 0}, {1, 1, 1}};
+
+    SECTION("Failure for wrong basisState size") {
+        std::size_t num_qubits = 3;
+        std::size_t maxBondDim = 3;
+        std::vector<size_t> basisState = {0, 0, 0, 0};
+
+        MPSCutn<TestType> mps_state{num_qubits, maxBondDim};
+
+        REQUIRE_THROWS_WITH(
+            mps_state.setBasisState(basisState),
+            Catch::Matchers::Contains("The size of a basis state should be "
+                                      "equal to the number of qubits."));
+    }
+
+    SECTION("Failure for wrong basisState input") {
+        std::size_t num_qubits = 3;
+        std::size_t maxBondDim = 3;
+        std::vector<size_t> basisState = {0, 0, 2};
+
+        MPSCutn<TestType> mps_state{num_qubits, maxBondDim};
+
+        REQUIRE_THROWS_WITH(
+            mps_state.setBasisState(basisState),
+            Catch::Matchers::Contains("Please ensure all elements of a basis "
+                                      "state should be either 0 or 1."));
+    }
+
+    SECTION("Set reset on device with data on the host") {
+        std::size_t bondDim = GENERATE(2, 3, 4, 5);
+        std::size_t num_qubits = 3;
+        std::size_t maxBondDim = bondDim;
+
+        MPSCutn<TestType> mps_state{num_qubits, bondDim};
+
+        mps_state.reset();
+
+        std::vector<std::complex<TestType>> expected_state(
+            size_t{1} << num_qubits, std::complex<TestType>({0.0, 0.0}));
+
+        std::size_t index = 0;
+
+        expected_state[index] = {1.0, 0.0};
+
+        CHECK(mps_state.getMaxBondDim() == maxBondDim);
+
+        CHECK(expected_state ==
+              Pennylane::Util::approx(mps_state.getDataVector()));
+    }
+
+    SECTION("Test different bondDim and different basisstate") {
+        std::size_t bondDim = GENERATE(2, 3, 4, 5);
+        for (auto &basisState : basisStates) {
+            std::size_t num_qubits = 3;
+            std::size_t maxBondDim = bondDim;
+
+            MPSCutn<TestType> mps_state{num_qubits, maxBondDim};
+
+            mps_state.setBasisState(basisState);
+
+            std::vector<std::complex<TestType>> expected_state(
+                size_t{1} << num_qubits, std::complex<TestType>({0.0, 0.0}));
```

Comment:
Isn't the following enough (above and under too)?
```suggestion
                size_t{1} << num_qubits, 0.0);
```

Reply:
`expected_state` should be `std::complex` data. Will keep current implementation